File: examples/R/example_dplyr.R
library("dplyr")
df <- MASS::Boston %>%
  filter(medv < 37) %>%
  select(rm, rad, lstat, medv) %>%
  mutate(medv1 = medv*1000) %>%
  arrange(desc(medv))
head(df)
gf <- df %>%
  group_by(rad) %>%
  summarise(
    ave_medv = mean(medv, na.rm=TRUE),
    .groups = 'drop'  # Prevents grouping in the final output
  )
gf

File: examples/Python/example_dplyr.py
import pandas as pd
df = r["MASS::Boston"]
df_filtered = (df[df['medv'] < 37]
               .loc[:, ['rm', 'rad', 'lstat', 'medv']]
               .assign(medv1 = df['medv'] * 1000)
               .sort_values(by='medv', ascending=False))
print(df_filtered.head())
df_grouped = df_filtered.groupby('rad'). \
               agg(ave_medv=('medv', 'mean')).reset_index()
print(df_grouped)
